_studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp

```diff
@@ -0,0 +1,1196 @@
+    vp.film_grain = par->mfx.FilmGrain ? 1 : 0; // 0 - film grain is forced off, 1 - film grain is controlled by apply_grain syntax parameter
+    if (!vp.async_depth)
+        vp.async_depth = MFX_AUTO_ASYNC_DEPTH_VALUE;
+    vp.io_pattern = par->IOPattern;
+
+#if defined (MFX_VA)
+    sts = m_core->CreateVA(par, &m_request, &m_response, m_allocator.get());
+    MFX_CHECK_STS(sts);
+
+    m_core->GetVA((mfxHDL*)&vp.pVideoAccelerator, MFX_MEMTYPE_FROM_DECODE);
+#endif
+
+    ConvertMFXParamsToUMC(par, &vp);
+
+    umcSts = m_decoder->Init(&vp);
+    MFX_CHECK(umcSts == UMC::UMC_OK, MFX_ERR_NOT_INITIALIZED);
+
+    m_first_run = true;
+    m_is_init = true;
+
+    return MFX_ERR_NONE;
+}
+
+// Check if new parameters are compatible with old parameters
+bool VideoDECODEAV1::IsNeedChangeVideoParam(mfxVideoParam * newPar, mfxVideoParam * oldPar, eMFXHWType /*type*/) const
+{
+    if ((newPar->IOPattern & (MFX_IOPATTERN_OUT_OPAQUE_MEMORY | MFX_IOPATTERN_OUT_SYSTEM_MEMORY | MFX_IOPATTERN_OUT_VIDEO_MEMORY)) !=
+        (oldPar->IOPattern & (MFX_IOPATTERN_OUT_OPAQUE_MEMORY | MFX_IOPATTERN_OUT_SYSTEM_MEMORY | MFX_IOPATTERN_OUT_VIDEO_MEMORY)))
+    {
+        return false;
+    }
+
+    if (CalculateAsyncDepth(m_platform, newPar) != CalculateAsyncDepth(m_platform, oldPar))
+    {
+        return false;
+    }
+
+    mfxFrameAllocRequest requestOld;
+    memset(&requestOld, 0, sizeof(requestOld));
+    mfxFrameAllocRequest requestNew;
+    memset(&requestNew, 0, sizeof(requestNew));
+
+    mfxStatus mfxSts = MFX_VPX_Utility::QueryIOSurfInternal(oldPar, &requestOld);
+
+    if (mfxSts != MFX_ERR_NONE)
+        return false;
+
+    mfxSts = MFX_VPX_Utility::QueryIOSurfInternal(newPar, &requestNew);
+
+    if (mfxSts != MFX_ERR_NONE)
+        return false;
+
+    if (newPar->mfx.FrameInfo.Height > oldPar->mfx.FrameInfo.Height)
+    {
+        return false;
+    }
+
+    if (newPar->mfx.FrameInfo.Width > oldPar->mfx.FrameInfo.Width)
+    {
+        return false;
+    }
+
+    if (m_response.NumFrameActual)
+    {
+        if (requestNew.NumFrameMin > m_response.NumFrameActual)
+            return false;
+    }
+    else
+    {
+        if (requestNew.NumFrameMin > requestOld.NumFrameMin || requestNew.Type != requestOld.Type)
+            return false;
+    }
+
+    if (newPar->mfx.FrameInfo.FourCC != oldPar->mfx.FrameInfo.FourCC)
+    {
+        return false;
+    }
+
+    if (newPar->mfx.FrameInfo.ChromaFormat != oldPar->mfx.FrameInfo.ChromaFormat)
+    {
+        return false;
+    }
+
+    if (oldPar->IOPattern & MFX_IOPATTERN_OUT_OPAQUE_MEMORY)
+    {
+        mfxExtOpaqueSurfaceAlloc * opaqueNew = (mfxExtOpaqueSurfaceAlloc *)GetExtendedBuffer(newPar->ExtParam, newPar->NumExtParam, MFX_EXTBUFF_OPAQUE_SURFACE_ALLOCATION);
+        mfxExtOpaqueSurfaceAlloc * opaqueOld = (mfxExtOpaqueSurfaceAlloc *)GetExtendedBuffer(oldPar->ExtParam, oldPar->NumExtParam, MFX_EXTBUFF_OPAQUE_SURFACE_ALLOCATION);
+
+        if (opaqueNew && opaqueOld)
```

Comment:
I'd better changed this condition to remove code folding:

```
if (!opaqueNew || !opaqueOld)
    return false;

...
```

Reply:
sure.